### DISCUSSION TITLE
Windows Terminal that uses Powershelll adds `"` at the end if a folder contains `\` at the end

### DISCUSSION THREAD

**Discussion by deadcoder0904** [PowerShell/PowerShell#21065] [Q&A]
Version: Deno v1.39.2

I faced this bug in `deno` using Windows 11 & thought it was a [Windows Terminal bug](https://github.com/microsoft/terminal/issues/16536) but others didn't think so I thought it must be [a bug with deno](https://github.com/denoland/deno/issues/21883) on Windows but they didn't think so either.

So finally I wanna see if its Powershell bug. Mind you I've faced this bug in Windows Terminal only that uses Powershell in 2 languages: python & deno.

### Windows Terminal version

Version: 1.18.3181.0

### Windows build number

Microsoft Windows [Version 10.0.22621.2861]

### Other Software

deno 1.39.2 (release, x86_64-pc-windows-msvc)
v8 12.0.267.8
typescript 5.3.3

### Steps to reproduce

Install [deno](https://deno.com/)

Paste the following script in a folder that has a subfolder.

### doublequote-bug.ts

```ts
// Get folder path from Deno.args
const [folderPath] = Deno.args

// Check if folderPath is provided
if (!folderPath) {
	console.error('Please provide the folder path as an argument.')
	Deno.exit(1)
}

const root = folderPath.replace(/"/g, '') // terminal bug that adds " if foldername contains / at the end

console.log({
	folderPath,
	root,
})
```

Now run it using deno.

### Expected Behavior

`folderPath` shouldn't have `"` at the end & it should look like `root`

This only happens when you have `\` at the end while calling `deno run`.

If I remove `\` manually after pressing `Tab` after typing `how<Tab>`, then both `folderPath` & `root` looks same.

```ts
PS D:\$$ Tuts> deno run .\doublequote-bug.ts '.\How to make $10k a month on Reddit'
{
  folderPath: ".\\How to make $10k a month on Reddit",
  root: ".\\How to make $10k a month on Reddit"
}
```

This is a source of many bugs that I faced in python as well as deno so I think it's actually a Terminal bug.

### Actual Behavior

Here's what I run & the output I get:

```ts
PS D:\$$ Tuts> deno run .\doublequote-bug.ts '.\How to make $10k a month on Reddit\'
{
  folderPath: '.\\How to make $10k a month on Reddit"',
  root: ".\\How to make $10k a month on Reddit"
}
```

Notice the extra double quote near the end of `folderPath`

**Comment by fMichaleczek** [comment]
@deadcoder0904 you forgot PowerShell version 😅
```
$PSVersionTable.PSVersion.ToString()
```
Can you try with latest PowerShell version (7.4.1) ?

**Reply by deadcoder0904** [reply]
i got `5.1.22621.2506`

> Can you try with latest PowerShell version (7.4.1) ?

bdw, what i did was default on windows 10 (pc) & 11 (laptop) so it should work regardless.

i updated to powershell 7 using https://lazyadmin.nl/powershell/update-powershell/ & i get what i want.

```ts
PS D:\$$ Tuts> deno run .\doublequote-bug.ts '.\How to make $10k a month on Reddit\'
{
  folderPath: ".\\How to make $10k a month on Reddit\\",
  root: ".\\How to make $10k a month on Reddit\\"
}
PS D:\$$ Tuts> deno run .\doublequote-bug.ts '.\How to make $10k a month on Reddit'
{
  folderPath: ".\\How to make $10k a month on Reddit",
  root: ".\\How to make $10k a month on Reddit"
}
```

no errors i guess.

i'm assuming you guys aren't gonna solve powershell 5 bug?

when will powershell 7 become default on os? has it already reached powershell 5 compatibility.

**Reply by deadcoder0904** [reply]
bdw i lost auto-up functionality.

like i had `z` installed & i used to do `z <up-arrow>` to find the particular `z <path>`

i tried looking at `.json` file in preferences but confused. know what its called? i think i had installed a module/plugin before like z & a bunch of others.

**Reply by fMichaleczek** [reply]
@deadcoder0904 No replacement date.

- Windows PowerShell is proprietary software that has been frozen since 2016 (8 years...)
- PowerShell is open source and its repo is here.
- 
Yes you need to migrate your modules

I'll give you the links to the official doc.
https://learn.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.4

https://learn.microsoft.com/en-us/powershell/scripting/whats-new/migrating-from-windows-powershell-51-to-powershell-7?view=powershell-7.4#separate-psmodulepath

**Reply by deadcoder0904** [reply]
those guides didn't give me the modules i wanted but did let me know the difference.

anyways, i found what i wanted from googling. i remember the author name was scott & added powershell to find his 2 articles & this one was the one i used:

https://www.hanselman.com/blog/adding-predictive-intellisense-to-my-windows-terminal-powershell-prompt-with-psreadline

i guess i'll be using powershell 7 from now as it does everything but better.
